drivers/docker/driver.go

```diff
@@ -1032,61 +1032,39 @@ func (d *Driver) createContainerConfig(task *drivers.TaskConfig, driverConfig *T
 	}
 
 	// Setup port mapping and exposed ports
-	if len(task.Resources.NomadResources.Networks) == 0 {
-		if len(driverConfig.PortMap) > 0 {
-			return c, fmt.Errorf("Trying to map ports but no network interface is available")
+	ports := newPublishedPorts(logger)
+	switch {
+	case task.Resources.Ports != nil && len(driverConfig.Ports) > 0:
```

Comment:
This supports the case where only one source of ports is specified?  Should we report an error, if we notice multiple sources as well?

Reply:
Idk if we should prevent the task from running since one of the fields is deprecated. It is documented in the docs that the two fields are incompatible.

Reply:
Shouldn't a deprecated incompatible spec result in the job spec being invalid? I can see this easily tripping up people. Or at least log a warning?

Reply:
We could log a warning but I do t think that would help the user if they have to go digging through client logs. With driver plugins, the driver configuration can’t be validated until runtime. There’s an open issue somewhere to explore fixing that but it’s an in-depth change.

So our only option is to return an error and fail the allocation which is totally reasonable. I can argue either way 😅

Reply:
Makes sense.

This is just my 5c as a user but I'd much rather have unsupported things fail on first attempt with an explanatory error message than scratching my head about why it's not working as intended ^^

Reply:
IIUC it's invalid if both group level network as well as task level network is specified?  Is that true?  If so, we can validate early on job submission regardless of driver?

For driver configuration warnings, we can also emit a TaskEvent, in leu of a log line.  It'll be visible with `nomad {alloc|job} status`, though I agree it's still not very discoverable.